src/Views/Dog/Dog.js

```diff
@@ -20,10 +23,19 @@ export default function Dog() {
   }, [id]);
   if (loading) return <h1>loading</h1>;
 
+  const deleteButton = async () => {
+    await deleteDog(dog.id);
+    setMessage('success');
+    setTimeout(() => {
```

Comment:
You might try to disable the form button's while the page waits to re-direct to avoid bugs